File: seed/csharp-model/trace/src/SeedTraceClient/Submission/ErrorInfo.cs
using SeedTraceClient
using System.Text.Json.Serialization

namespace SeedTraceClient

public class ErrorInfo
{
    public class _CompileError : CompileError
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "compileError";
    }
    public class _RuntimeError : RuntimeError
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "runtimeError";
    }
    public class _InternalError : InternalError
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "internalError";
    }
}

File: seed/rust-sdk/trace/src/types/error_info.rs
use crate::compile_error::CompileError;
use crate::runtime_error::RuntimeError;
use crate::internal_error::InternalError;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum ErrorInfo {
        CompileError {
            #[serde(flatten)]
            data: CompileError,
        },

        RuntimeError {
            #[serde(flatten)]
            data: RuntimeError,
        },

        InternalError {
            #[serde(flatten)]
            data: InternalError,
        },
}
